mobile-ui/src/main/java/com/playone/mobile/ui/playone/PlayoneActivity.kt

```diff
@@ -25,15 +28,48 @@ class PlayoneActivity : BaseActivity() {
 
         viewModel = ViewModelProviders.of(this, viewModelFactory)
             .get(LoginViewModel::class.java)
+        viewModel.isProgressing.observe(this, Observer {
+            it?.takeIf { it }?.apply { showProgress() } ?: hideProgress()
+        })
+        viewModel.isSignedIn.observe(this, Observer {
+            it?.takeIf { it }?.apply {
+                // TODO: proceed to the main page with logged in state
+            } ?: showSignInForms()
+        })
+        viewModel.occurredError.observe(this, Observer {
+            it?.takeIf { it.isNotNull() }?.apply { showErrorState(it) }
+        })
+        viewModel.let { lifecycle.addObserver(it) }
 
-        // TODO: Should remove this. Just for demo now
-        Handler().postDelayed(3000L) {
-            showSignInForms()
-        }
+        viewModel.isSignedIn()
     }
 
     private fun showSignInForms() {
+        val view = layoutInflater.inflate(R.layout.merge_login, content_layout, true)
+
+        // Sign in with email and password
+        val emailEditText = view.findViewById<EditText>(R.id.login_name_field)
```

Comment:
There is a better way for getting a view from the XML file by Kotlin extension.

```kotlin
view.login_name_field
```
I always do as like this and do really like it! How about it?

Reply:
LTGM